# HG changeset patch
# User Rik <rik@octave.org>
# Date 1376595082 25200
#      Thu Aug 15 12:31:22 2013 -0700
# Node ID ee1d19174316b38e9487eec84e770ca09c8c27b5
# Parent  a28c0d73e2532c0ce1f7de54763bedc9f77e5694
doc: Use 0x1d as record separator for joint compatibility with Texinfo 4.x and 5.x.

* doc/interpreter/mk_doc_cache.m, doc/interpreter/munge-texi.pl,
libinterp/corefcn/help.cc(install_built_in_docstrings), libinterp/gendoc.pl,
scripts/mkdoc.pl: Use 0x1d as record separator.

diff --git a/doc/interpreter/mk_doc_cache.m b/doc/interpreter/mk_doc_cache.m
--- a/doc/interpreter/mk_doc_cache.m
+++ b/doc/interpreter/mk_doc_cache.m
@@ -19,17 +19,17 @@ args = argv ();
 if (nargin < 2)
   error ("usage: mk_doc_cache OUTPUT-FILE DOCSTRINGS-FILE ...");
 endif
 
 output_file = args{1};
 docstrings_files = args(2:end);
 
 ## Special character used as break between DOCSTRINGS
-doc_delim = char (0x1e);
+doc_delim = char (0x1d);
 
 ## Read the contents of all the DOCSTRINGS files into TEXT.
 ## It is more efficient to fork to shell for makeinfo only once on large data
 
 nfiles = numel (docstrings_files);
 text = cell (1, nfiles);
 for i = 1:nfiles
   file = docstrings_files{i};
diff --git a/doc/interpreter/munge-texi.pl b/doc/interpreter/munge-texi.pl
--- a/doc/interpreter/munge-texi.pl
+++ b/doc/interpreter/munge-texi.pl
@@ -1,17 +1,17 @@
 #!/usr/bin/perl -w
 
 # Validate program call
 die "usage: munge-texi TOP-SRCDIR DOCSTRING-FILE < file" if (@ARGV < 2);
 
 $top_srcdir = shift (@ARGV);
 
 # Constant patterns
-$doc_delim = qr/^\x{1e}/;
+$doc_delim = qr/^\x{1d}/;
 $tex_delim = qr/\Q-*- texinfo -*-\E/;
 $comment_line = qr/^\s*(?:$|#)/;
 # Pre-declare hash size for efficiency
 keys(%help_text) = 1800;
 
 ################################################################################
 # Load DOCSTRINGS into memory while expanding @seealso references
 foreach $DOCSTRING_file (@ARGV)
diff --git a/libinterp/corefcn/help.cc b/libinterp/corefcn/help.cc
--- a/libinterp/corefcn/help.cc
+++ b/libinterp/corefcn/help.cc
@@ -962,17 +962,17 @@ install_built_in_docstrings (void)
 {
   std::string fname = Vbuilt_in_docstrings_file;
 
   std::ifstream file (fname.c_str (), std::ios::in | std::ios::binary);
 
   if (file)
     {
       // Ignore header;
-      file.ignore (1000, 0x1e);
+      file.ignore (1000, 0x1d);
 
       if (file.gcount () == 1000)
         {
           // We use std::cerr here instead of calling Octave's warning
           // function because install_built_in_docstrings is called
           // before the interpreter is initialized, so warning messages
           // won't work properly.
 
@@ -983,17 +983,17 @@ install_built_in_docstrings (void)
 
       // FIXME -- eliminate fixed buffer size.
       size_t bufsize = 100000;
 
       OCTAVE_LOCAL_BUFFER (char, buf, bufsize);
 
       while (! file.eof ())
         {
-          file.getline (buf, bufsize, 0x1e);
+          file.getline (buf, bufsize, 0x1d);
 
           std::string tmp (buf);
 
           size_t pos = tmp.find ('\n');
 
           std::string fcn = tmp.substr (0, pos);
 
           octave_value ov = symbol_table::find_built_in_function (fcn);
diff --git a/libinterp/gendoc.pl b/libinterp/gendoc.pl
--- a/libinterp/gendoc.pl
+++ b/libinterp/gendoc.pl
@@ -59,15 +59,15 @@ DFFILE: foreach $df_fname (@ARGV)
     }
   }
   close (DF_FH);
 
   ## Print results in DOCSTRING format
   foreach $i (0 .. $#func_list)
   {
     $func = $func_list[$i];
-    print "\x{1e}$func\n";
+    print "\x{1d}$func\n";
     print "\@c $func $src_fname\n";
     print $docstr[$i],"\n";
   }
 
 }
 
diff --git a/scripts/mkdoc.pl b/scripts/mkdoc.pl
--- a/scripts/mkdoc.pl
+++ b/scripts/mkdoc.pl
@@ -47,17 +47,17 @@ MFILE: foreach $m_fname (@ARGV)
   if ($2)
     { $fcn = "$2$3/$4"; }
   else
     { $fcn = $4; }
 
   @help_txt = gethelp ($fcn, $full_fname);
   next MFILE if ($help_txt[0] eq "");
 
-  print "\x{1e}$fcn\n";
+  print "\x{1d}$fcn\n";
   print "\@c $fcn scripts/$m_fname\n";
 
   foreach $_ (@help_txt)
   {
     s/^\s+\@/\@/ unless $in_example;
     s/^\s+\@group/\@group/;
     s/^\s+\@end\s+group/\@end group/;
     $in_example = (/\s*\@example\b/ .. /\s*\@end\s+example\b/);
